treelib/tree.py

```diff
@@ -801,23 +802,23 @@ def show(self, nid=None, level=ROOT, idhidden=True, filter=None,
         :param reverse: the ``reverse`` param for sorting :class:`Node` objects in the same level.
         :param line_type:
         :param data_property: the property on the node data object to be printed.
+        :param end: the string to append to the final line of the output. By
+            default, an extra newline is printed; pass '\n' if this is not
+            desired.
         :return: None
         """
-        self._reader = ""
+        self._reader = []
 
         def write(line):
-            self._reader += line.decode('utf-8') + "\n"
+            self._reader.append(line.decode('utf-8'))
 
         try:
             self.__print_backend(nid, level, idhidden, filter,
                                  key, reverse, line_type, data_property, func=write)
         except NodeIDAbsentError:
             print('Tree is empty')
 
-        if stdout:
-            print(self._reader)
-        else:
-            return self._reader
+        print('\n'.join(self._reader), end=end)
```

Comment:
I strongly disagree with using `print`, see https://github.com/caesar0301/treelib/issues/141

 Using `print` to `stdout` in a library is a bad practice, see for instance [this article](https://timber.io/blog/the-pythonic-guide-to-logging/) 
>  It's also bad practice to ship a package that prints directly to stdout because it removes the user's ability to control the messages.

Reply:
This article talk about logging. I don't think It is related to this use case here. Here it is just an helper function.

You could create 2 functions, it would make the API simpler. I also think a stream as input is better, anyway the default value is stdout.